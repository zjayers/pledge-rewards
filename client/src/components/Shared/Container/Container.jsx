import React from 'react';
import Container from './Container.style';

export default (props) => {
  const { activated, padding } = props;
  const { children } = props;
  return (
    <Container activated={activated} padding={padding}>
      {children}
    </Container>
  );
};
