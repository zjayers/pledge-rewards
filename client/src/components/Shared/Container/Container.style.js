import styled from 'styled-components';

export default styled.div`
  width: 100%;
  border: 1px solid ${(props) => props.theme.colors.borderGrey};
  display: flex;
  flex-direction: column;
  padding: ${(props) => props.padding} 20px 20px;
  margin-bottom: 20px;
  transition: all 0.2s ease-in-out;
  position: relative;
`;
