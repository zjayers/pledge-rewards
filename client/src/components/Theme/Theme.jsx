import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
