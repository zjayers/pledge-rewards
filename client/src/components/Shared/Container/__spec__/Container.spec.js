import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Container from '../Container';
import Theme from '../../../Theme/Theme';

describe('Container Component', () => {
  let component;

  beforeEach(() => {
    component = mount(
      <Theme>
        <Container />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
