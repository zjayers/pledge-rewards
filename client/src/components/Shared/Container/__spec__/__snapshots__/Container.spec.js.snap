// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <_default>
      <Containerstyle___default>
        <div
          className="Containerstyle___default-sc-1q8wjhx-0 BmWDh"
        />
      </Containerstyle___default>
    </_default>
  </ThemeProvider>
</Theme>
`;
